improver/utilities/temporal.py

```diff
@@ -281,3 +281,67 @@ def relabel_to_period(cube: Cube, period: Optional[int] = None):
             dtype=TIME_COORDS[coord].dtype,
         )
     return cube
+
+
+def integrate_time(cube: Cube, new_name: str = None) -> Cube:
+    """
+    Multiply a frequency or rate cube by the time period given by the
+    time bounds over which it is defined to return a count or accumulation.
+    The frequency or rate must be defined with time bounds, e.g. an average
+    frequency across the period.
+
+    The returned cube has units equivalent to the input cube multiplied by
+    seconds.
+
+    Any time related cell methods are removed from the output cube and a new
+    "sum" over time cell method is added.
+
+    Args:
+        Cube:
+            A cube of average frequency or rate within a defined period.
+        new_name:
+            A new name for the resulting diagnostic.
+
+    Returns:
+        The cube with the data multiplied by the period in seconds defined
+        by the time time bounds
+
+    Raises:
+        ValueError: If the input cube time coordinate does not have time
+                    bounds.
+    """
+    # Ensure cube has a time coordinate with bounds
+    if not cube.coord("time").has_bounds():
+        raise ValueError(
+            "time coordinate must have bounds to apply this time-bounds " "integration"
+        )
+
+    # For each grid of data associated with a time, multiply the rate / frequency
+    # by the associated time interval to get an accumulation / count over the
+    # period.
+    integrated_cube = iris.cube.CubeList()
+    for cslice in cube.slices_over("time"):
+        (multiplier,) = np.diff(cslice.coord("time").cell(0).bound)
+        multiplier = multiplier.total_seconds()
+        cslice.data *= multiplier
+        integrated_cube.append(cslice)
+
+    integrated_cube = integrated_cube.merge_cube()
+
+    # Modify the cube units to reflect the multiplication by time.
+    integrated_cube.units *= cf_units.Unit("s")
+    if new_name is not None:
+        integrated_cube.rename(new_name)
```

Comment:
Just wondering, about whether `new_name` should be optional, or whether this should be expected given that the variable will be different following the conversion from frequency to accumulation? I'm happy to leave this for the moment, as I'm not sure about all potential use cases for this functionality, but I thought that I'd ask.